ospfd/ospf_packet.c

```diff
@@ -2685,6 +2685,13 @@ static unsigned ospf_lsa_examin(struct lsa_header *lsah, const uint16_t lsalen,
 	case OSPF_OPAQUE_LINK_LSA:
 	case OSPF_OPAQUE_AREA_LSA:
 	case OSPF_OPAQUE_AS_LSA:
+		if (!headeronly && (lsalen - OSPF_LSA_HEADER_SIZE) == 0) {
```

Comment:
I don't think this is the right fix.
There's already a length check at the beginning of this function:
```
	if (lsah->type < OSPF_MAX_LSA && ospf_lsa_minlen[lsah->type]
	    && lsalen < OSPF_LSA_HEADER_SIZE + ospf_lsa_minlen[lsah->type]) {
		if (IS_DEBUG_OSPF_PACKET(0, RECV))
			zlog_debug("%s: undersized (%u B) %s", __func__, lsalen,
				   lookup_msg(ospf_lsa_type_msg, lsah->type,
					      NULL));
		return MSG_NG;
	}
```

If you think some of the LSA types must have a different minimum length, then you need to populate `ospf_lsa_minlen` array with the correct values instead of adding this additional check.
By the way, I didn't find any restrictions on the length of opaque LSAs in RFCs. Do you have a specific issue in mind that you are fixing?

Reply:
I am of two minds on this one ... I agree that receiving an LSA with no data should just be a "no-op." OTOH, I do think we should flag/log such a condition, if for no other reason than to tell the operator to go ask the implementer sending these no-length LSA's "what's up?" 

As for how to do the check, I'm ambivalent. :-)

Reply:
Like Russ, I think we must log this in case of its happen. Now regarding the fix, I agree with Igor that there is a check on the LSA size at the beginning of the function. Now, regarding RFC, there is no minimum length for Opaque LSA defined. But, if we look to some specific RFCs that use Opaque LSA to carry information, at the very minimum, 4 bytes must be present. Thus, the correct patch would be to add a new `#define OSPF_OPAQUE_LSA_MIN_SIZE 4U` in `ospf_lsa.h` and fulfill the `ospf_lsa_minlen` table in `ospf_packet.c` lines 111, 112 & 113 (for the 3 type of Opaque LSA) with that value.